shell/imports/client/grain/grainlist-client.js

```diff
@@ -270,6 +271,28 @@ SandstormGrainListPage.bulkActionButtons = function (showTrash) {
           Meteor.call("moveGrainsToTrash", ownedGrainIds.concat(sharedGrainIds));
         },
       },
+      {
+        buttonClass: "save-backup",
+
+        text: function() {
+          return 'Save backup';
+        },
+
+        disabled: function(numMineSelected, numSharedSelected) {
+          return numMineSelected === 0 || numSharedSelected !== 0;
+          // Users can only back up their own grains, so disable if
+          // they've selected anyone else's. TODO: give the user more
+          // feedback on this somehow.
+        },
+
+        onClicked: function (ownedGrainIds, _sharedGrainIds) {
+          ownedGrainIds.forEach((grainId) => {
+            // TODO: use the grain title, not the grain id, as the suggested
+            // file name.
+            makeAndDownloadBackup(grainId, grainId);
```

Comment:
I don't understand why, but in testing this actually does use the grain's title as the file name -- so perhaps the browser is ignoring the hint we have here anyway?